File: TypeScript/Katas/cardMask.ts
// Codewars (Vanilla JS Kata done in TS)

/*===================================================================================================
    Your task is to write a function maskify, which changes all but the last four characters into '#'.
===================================================================================================*/

export function maskify(cc: string): string
{
    let ccArray = cc.split('');
    let maskedList: string[] = [];

    for (let i = 0; i < ccArray.length; i++) {
        if (i < (ccArray.length - 4)) {
            maskedList.push("#");
        } else {
            maskedList.push(ccArray[i].toString());
        }
    }

    return maskedList.join("");
}

export function maskify2ElectricBoogaloo(cc: string): string
{
    return cc.replace(/.(?=.{4})/g, "#");
}

File: C#/Katas/KataClasses/CardMask.cs
// Codewars

/*===================================================================================================
    Your task is to write a function maskify, which changes all but the last four characters into '#'.
===================================================================================================*/

namespace Katas.KataClasses;

public class CardMask
{
    public static string Maskify(string cc)
    {
        char[] ccArray = cc.ToCharArray();
        List<String> maskedList = new List<String> { };

        for (int i = 0; i < ccArray.Length; i++)
        {
            if (i < (ccArray.Length - 4))
                maskedList.Add("#");
            else
                maskedList.Add(ccArray[i].ToString());
        }

        return string.Join("", maskedList);

        // Using Regex.Replace(cc, {REGEX HERE}, "#") would have worked too.
    }
}